BitwardenShared/UI/Platform/Application/Utilities/Alert/Alert/Alert.swift

```diff
@@ -99,39 +99,68 @@ public class Alert {
     ///
     @MainActor
     func createAlertController(onDismissed: (() -> Void)? = nil) -> UIAlertController {
-        let alert = AlertController(title: title, message: message, preferredStyle: preferredStyle)
-        alert.onDismissed = onDismissed
-        alertTextFields.forEach { alertTextField in
-            alert.addTextField { textField in
-                textField.placeholder = alertTextField.placeholder
-                textField.keyboardType = alertTextField.keyboardType
-                textField.isSecureTextEntry = alertTextField.isSecureTextEntry
-                textField.autocapitalizationType = alertTextField.autocapitalizationType
-                textField.autocorrectionType = alertTextField.autocorrectionType
-                textField.text = alertTextField.text
+        let alertController = AlertController(title: title, message: message, preferredStyle: preferredStyle)
+        alertController.onDismissed = onDismissed
+
+        let shouldUpdateActions = alertActions.contains { $0.shouldEnableAction != nil }
+
+        addTextFields(to: alertController, updateActionsIfNeeded: shouldUpdateActions)
+        addActions(to: alertController)
+
+        return alertController
+    }
+
+    private func addTextFields(to alertController: UIAlertController, updateActionsIfNeeded: Bool) {
+        for alertTextField in alertTextFields {
+            alertController.addTextField { textField in
+                self.configure(textField, with: alertTextField)
+
                 textField.addTarget(
                     alertTextField,
                     action: #selector(AlertTextField.textChanged(in:)),
                     for: .editingChanged
                 )
             }
+
+            if updateActionsIfNeeded {
+                alertTextField.onTextChanged = { [weak self, weak alertController] in
+                    guard let self, let alert = alertController else { return }
```

Comment:
🎨 I think you can directly use `alertController` without assigning to `alert` here:
```suggestion
                    guard let self, let alertController else { return }
```